pandas/io/html.py

```diff
@@ -637,41 +688,30 @@ def _build_doc(self):
                 raise XMLSyntaxError("no text parsed from document", 0, 0, 0)
         return r
 
-    def _parse_tbody(self, table):
-        return table.xpath('.//tbody')
+    def _parse_thead_tr(self, table):
+        rows = []
 
-    def _parse_thead(self, table):
-        return table.xpath('.//thead')
+        for thead in table.xpath('.//thead'):
+            rows.extend(thead.xpath('./tr'))
 
-    def _parse_tfoot(self, table):
-        return table.xpath('.//tfoot')
+            # lxml does not clean up the clearly-erroneous
+            # <thead><th>foo</th><th>bar</th></thead>.
```

Comment:
Probably my lack of understanding but what is erroneous about this? Missing `tr`?

Reply:
added comment, thanks